enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_any?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count
+    i = 0
+    count = []
+    if block_given?
+      while i < size
+        count << self[i] if yield(self[i])
+        i += 1
+      end
+      return count.size
+    end
+    size
+  end
```

Comment:
- my_count should count the number of items in enum that are equal to item if an argument is given
`[1,2,3,3,3].count (3)`
`=> 3`
`[1,2,3,3,3].count (1)`
`=> 1`